### PR TITLE
Re-Feature: Spending by Category

### PR DESCRIPTION
**Pull request by dpb-bah**
**High level description:**
Reviving the Spending by Category endpoint to be used on the Award Search page and State Profile Top 5 endpoints.

**Technical details:**
* Updated several matviews
* Fixed a bug for the `award_amount` filter
* Added extra mocks fixtures for testing
* Fixed award type filter bug (see `no intersection` value)
* TinyShield updates
* Updated Spending by Category Tests
* Moved Spending By Category to a Separate File, separated the Business Logic out for testing

**Link to JIRA Ticket:**
[DEV-876](https://federal-spending-transparency.atlassian.net/browse/DEV-876)

**The following are ALL required for the PR to be merged:**
- [ ] Backend review completed
- [x] Unit & integration tests updated with relevant test cases
- [x] Matview impact assessment completed
```
Removed cfda popular name from matviews
Added recipient data to matviews
Adding to subawardsview
	* abbrevations
	* cfda_title
```
- [x] Frontend impact assessment completed
- Data validation completed N/A
- [x] API Performance evaluation completed and present
```
{{url}}/api/v2/search/spending_by_category/, connected to staging
{
  "filters": {	
    "time_period": [
      {
        "start_date": "2016-10-01",
        "end_date": "2017-09-30",
        "date_type": "action_date"
      }
	]
  },
  "limit": 10,
  "page": 1
}

awarding_agency: 3103ms
awarding_subagency: 3471ms
awarding_agency (subawards): 4425ms
awarding_subagency (subawards): 3981ms
funding_agency: 3179ms
funding_subagency: 3254ms
funding_agency (subawards): 3968ms
funding_subagency (subawards): 4524ms
recipient_duns: 15659ms
recipient_duns (subawards): 15659ms
cfda: 5056ms
cfda (subawards): 4065ms
psc: 3849ms
naics: 2148ms
county: 3660ms
county (subawards): 2494ms
district: 2764ms
district (subawards): 4294ms
```

### CODE REVIEW

**Comment by kevinli-work** [comment]
Reverse link to frontend: fedspendingtransparency/usaspending-website#789

File: usaspending_api/search/v2/views/search.py [hunk 1]
```diff
@@ -644,6 +381,8 @@ def post(self, request):
         for m in models:
             if m['name'] in ('award_type_codes', 'fields'):
                 m['optional'] = False
+            if m['name'] == 'award_type_codes':
+                m['enum_values'] = m['enum_values'] + ['no intersection']  # handles an edge case from the Frontend
```
Comment (tony-sappe) [inline usaspending_api/search/v2/views/search.py]: Let's move 'no intersection' back to the award_type_codes model in usaspending_api/core/validator/award_filter.py

File: usaspending_api/api_docs/api_documentation/advanced_award_search/spending_by_category.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
+## [Spending by Award](#spending-by-category)
+**Route:** `/api/v2/search/spending_by_category/`
+
+**Method:** `POST`
+
+These endpoints return data that is grouped in preset units to support the various data visualizations on USAspending.gov's Advanced Search page.
+### Request
+
+filters: The filters to find with said category. The filter object is defined here: [Filter Object](../search_filters.md).
```
Comment (michaeldhess) [inline usaspending_api/api_docs/api_documentation/advanced_award_search/spending_by_category.md]: Add `(**REQUIRED**)` or `(**OPTIONAL**)` to this parameter as well.

File: usaspending_api/awards/v2/filters/view_selector.py [hunk 3]
```diff
@@ -308,3 +311,42 @@ def transaction_spending_summary(filters):
         raise InvalidParameterException
 
     return queryset, model
+
+
+def spending_by_category(category, filters):
+    # category is a string of <category>-<scope>.
```
Comment (michaeldhess) [inline usaspending_api/awards/v2/filters/view_selector.py]: This comment is wrong, category does not include `-scope`.

File: usaspending_api/database_scripts/matview_generator/matview_sql_generator.py [hunk 4]
```diff
@@ -297,11 +297,17 @@ def main(source_file):
...
         print(e)
-        raise SystemExit
+        raise SystemExit(1)
     create_monolith_file(sql_json)
     create_componentized_files(sql_json)
-    print('Done')
+    printf('Done')
+
+
+def printf(msg):
```
Comment (michaeldhess) [inline usaspending_api/database_scripts/matview_generator/matview_sql_generator.py]: `printf` is a default python class, maybe name this something else?

File: usaspending_api/search/tests/test_spending_by_category.py [hunk 5]
```diff
@@ -3,92 +3,51 @@
...
 from rest_framework import status
 
+from usaspending_api.search.tests.test_mock_data_search import all_filters
+
 
-@pytest.mark.skip
 @pytest.mark.django_db
-def test_spending_by_category_success(client):
+def test_spending_by_category_success(client, refresh_matviews):
+    print('donnnne')
```
Comment (michaeldhess) [inline usaspending_api/search/tests/test_spending_by_category.py]: Remove this print statement.

File: usaspending_api/awards/v2/filters/view_selector.py [hunk 6]
```diff
@@ -314,7 +314,7 @@ def transaction_spending_summary(filters):
 
 
 def spending_by_category(category, filters):
-    # category is a string of <category>-<scope>.
+    # category is a string of <category>.
     # It isn't elegant but it works enough until we can refactor this endpoint into individual endpoints
```
Comment (michaeldhess) [inline usaspending_api/awards/v2/filters/view_selector.py]: I don't think this comment applies to this function.

File: usaspending_api/search/v2/views/spending_by_category.py [hunk 7]
```diff
@@ -224,6 +221,9 @@ def industry_and_other_codes(self) -> list:
             filters = {'{}__isnull'.format(self.obligation_column): False, 'cfda_number__isnull': False}
             values = ['cfda_number']
         elif self.category == 'psc':
+            if self.subawards:
+                # TODO: get subaward NAICS from Broker
```
Comment (michaeldhess) [inline usaspending_api/search/v2/views/spending_by_category.py]: Is this TODO for the current Sprint or the future? Is there a JIRA ticket?

**Pull request merged**